Floofbot/Services/WordFilterService.cs

```diff
@@ -30,16 +30,21 @@ class WordFilterService
                 _lastRefreshedTime = currentTime;
             }
 
+            List<string> DetectedWords = new List<string>();
+
             foreach (var filteredWord in _filteredWords)
             {
                 Regex r = new Regex($"{filteredWord.Word}",
                     RegexOptions.IgnoreCase | RegexOptions.Singleline);
                 if (r.IsMatch(messageContent))
                 {
-                    return true;
+                    DetectedWords.Add(filteredWord.Word);
                 }
             }
-            return false;
+            if (DetectedWords.Count() == 0)
+                return null;
```

Comment:
While this would work, I'd say it's generally more favorable to return an empty list as it's conceptually more correct (i.e. the number of filtered words is zero).